# Customer Churn Prediction & Analysis

## Business Problem

Customer churn is a major challenge for telecom companies because losing customers directly impacts revenue and profitability. Acquiring a new customer is significantly more expensive than retaining an existing one.

The objective of this project is to analyze customer behavior, identify factors associated with churn, and develop a predictive model that helps the business proactively retain at-risk customers.

In [4]:
import os

print(os.listdir('../data/raw'))

['WA_Fn-UseC_-Telco-Customer-Churn.csv']


In [11]:
import pandas as pd

# Load dataset
df = pd.read_csv('../data/raw/WA_Fn-UseC_-Telco-Customer-Churn.csv')

print("Dataset loaded successfully!")
print("Dataset Shape:", df.shape)

Dataset loaded successfully!
Dataset Shape: (7043, 21)


In [12]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)

## Dataset Loading

The IBM Telco Customer Churn dataset was loaded into a Pandas DataFrame for analysis.

This dataset contains customer demographics, account information, subscribed services, billing details, and churn status.

In [13]:
# Load dataset

df = pd.read_csv('../data/raw/WA_Fn-UseC_-Telco-Customer-Churn.csv')

print("Dataset loaded successfully.")

Dataset loaded successfully.


## Dataset Size

To understand the scope of the analysis, I first checked the number of records and features available in the dataset.

In [14]:
print(f"Rows: {df.shape[0]}")
print(f"Columns: {df.shape[1]}")

Rows: 7043
Columns: 21


### Observation

The dataset contains 7,043 customer records and 21 columns. Each row represents an individual customer and each column contains information that may help explain customer churn.

## Feature Review

I reviewed the available columns to understand the customer information available for analysis and identify the target variable.

In [15]:
df.columns.tolist()


['customerID',
 'gender',
 'SeniorCitizen',
 'Partner',
 'Dependents',
 'tenure',
 'PhoneService',
 'MultipleLines',
 'InternetService',
 'OnlineSecurity',
 'OnlineBackup',
 'DeviceProtection',
 'TechSupport',
 'StreamingTV',
 'StreamingMovies',
 'Contract',
 'PaperlessBilling',
 'PaymentMethod',
 'MonthlyCharges',
 'TotalCharges',
 'Churn']

### Observation

The dataset includes customer demographics, account information, subscribed services, billing details, and churn status.
